### ISSUE TITLE
I use libtorch to run forward on Windows 10 times slower than on Ubuntu

### ISSUE THREAD

**Issue by zhongqingyang** [pytorch/pytorch#53943]
I loaded the same model file, used the same code and the same libtorch version（1.8.0）. Forward transmission is performed on CPU, but windows uses 5S, and Ubuntu only uses 500ms. What should I do in windows to improve the speed? Thank you very much for your help

cc @VitalyFedyunin @ngimel @peterjc123 @maxluk @nbcsm @guyang3532 @gunandrose4u @mszhanyi @skyline75489

**Comment by ngimel** [comment]
Please provide self contained code sample that can reproduce the problem.

**Comment by zhongqingyang** [comment]
int loadModel()
{
       device_type = at::kCPU;
    // 加载抠图模型
    try {
        model = torch::jit::load("u2net.pt");  //加载模型
    }
    catch (const c10::Error& e) {
        cout << "无法加载模型" << endl;
        return 0;
    }
    model.eval();
    model.to(device_type);
}
cv::Mat change_bg(cv::Mat des)
{

    Mat img = std::move(des);

    Mat orgimg = img.clone();

    // 压缩图像
    int org_width = img.cols;
    int org_height = img.rows;
    resize(img, img, Size(320, 320), 0, 0, INTER_CUBIC);

    //数据预处理
    std::vector<int64_t> sizes = { 1, img.rows, img.cols,3 };
    at::TensorOptions options(at::ScalarType::Byte);
    at::Tensor tensor_image = torch::from_blob(img.data, at::IntList(sizes), options);//将opencv的图像数据转为Tensor张量数据
    tensor_image = tensor_image.toType(at::kFloat);//转为浮点型张量数据

    tensor_image = tensor_image.permute({ 0, 3, 1, 2 });
    tensor_image[0][0] = tensor_image[0][0].sub(114).div(255.0);
    tensor_image[0][1] = tensor_image[0][1].sub(121).div(255.0);
    tensor_image[0][2] = tensor_image[0][2].sub(134).div(255.0);

    tensor_image = tensor_image.to(device_type);

    // 推理
    std::vector<torch::jit::IValue> inputs;
    inputs.emplace_back(tensor_image);
    at::Tensor out_tensor = model.forward(inputs).toTuple()->elements()[0].toTensor();

    out_tensor = out_tensor.squeeze().detach();
    out_tensor = out_tensor.mul(255).add_(0.5).clamp_(0, 255).to(torch::kU8);
    out_tensor = out_tensor.to(torch::kCPU);

    cv::Mat alphaImg(img.rows, img.cols, CV_8UC1);
    std::memcpy((void*)alphaImg.data, out_tensor.data_ptr(), sizeof(torch::kU8) * out_tensor.numel());
}
it is my code.


there is my model
https://drive.google.com/file/d/1DTHBCrWlDCYpoO9R8T5qr-TvhmcuyHH5/view?usp=sharing

**Comment by malfet** [comment]
Can you also elaborate what build systems (I presume CMake) and binary builds are using using to generate the binary?

**Comment by zhongqingyang** [comment]
[CMakeLists.txt](https://github.com/pytorch/pytorch/files/6145558/CMakeLists.txt)
This is myCMakeLists.txt，I used vs2019 in windows.
https://drive.google.com/file/d/1tQZEBlSJ_lGd2Wqu45DpXB4njukAhxpx/view?usp=sharing
Here are my conversion program and binary files，I tried u2net.pth And u2netp.pth Two models.
Thank you again for your help

**Comment by mszhanyi** [comment]
could you add your opencvdemo.cpp and opencvdemo.h

**Comment by zhongqingyang** [comment]
[opencvDemo.zip](https://github.com/pytorch/pytorch/files/6145841/opencvDemo.zip)
here are my opencvdemo.cpp and opencvdemo.h

**Comment by mszhanyi** [comment]
I tested in my PC. CPU Maximum speed: 2.6 GHz, virtual processors 8 core, windows 10
The Inference time is less than 1.1 seconds with OMP_NUM_THREADS=8.

**Comment by skyline75489** [comment]
@zhongqingyang Please make sure you're using the "release" configuration on Windows.

**Comment by zhongqingyang** [comment]
I uses cmake can't release,I change Release can faster,Thank you for your help.

**Comment by skyline75489** [comment]
I've tried the code myself and found "release" is about 10x faster than "debug", which is expected. Feel free to close the issue.
